src/utils/ucc_math.h

```diff
@@ -12,4 +12,31 @@
 #define ucc_min(_a, _b) ucs_min((_a), (_b))
 #define ucc_max(_a, _b) ucs_max((_a), (_b))
 
+static inline size_t ucc_dt_size(ucc_datatype_t dt)
```

Comment:
i remember you said it is better to use array map instead of switch branching. can't we use it here?

Reply:
i still don't remember why we haven't defined datatype as uint64_t like UCX. In that case we could actually pack the size of the predefined datatype in the higher bits of the uint64_t and don't need any map/switch at all. hcoll does it this way as well btw. @manjugv i remember i was making a proposal like that. what was the reason to discard it ?

Reply:
As I mentioned in our earlier conversation, let's bring this as separate PR and discuss. Very likely, it was not a standalone PR and did not get the attention it deserves.

Reply:
I've added array instead of switch in ucc_dt_size as a temporary solution